src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -60,27 +67,42 @@ public static CollectorManager createHybridCollectorManager(final SearchContext
         int numDocs = Math.min(searchContext.from() + searchContext.size(), totalNumDocs);
         int trackTotalHitsUpTo = searchContext.trackTotalHitsUpTo();
 
+        Weight filteringWeight = null;
+        // Check for post filter to create weight
+        if (Objects.nonNull(searchContext.parsedPostFilter())) {
+            Query filterQuery = searchContext.parsedPostFilter().query();
+            ContextIndexSearcher searcher = searchContext.searcher();
+            filteringWeight = searcher.createWeight(searcher.rewrite(filterQuery), ScoreMode.COMPLETE_NO_SCORES, 1f);
```

Comment:
why these params are used for create weight: COMPLETE_NO_SCORES score mode, and boost factor 1? Additionally, boost factor can be referenced via some constant I believe.

Reply:
ScoreMode COMPLETE_NO_SCORES will be passed as post_filter does not contribute in scoring. COMPLETE_NO_SCORES means it is not a scoring clause. Boost factor 1f is taken because if boost is multiplicative of 1 then it means "no boost". Previously this code in OpenSearch looked like https://github.com/opensearch-project/OpenSearch/commit/36a5cf8f35e5cbaa1ff857b5a5db8c02edc1a187